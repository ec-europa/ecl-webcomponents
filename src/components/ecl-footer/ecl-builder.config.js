const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  copy: [
    {
      from: [`${nodeModules}/@ecl/resources-ec-logo/dist/**/*.svg`],
      to: path.resolve(outputFolder, 'images/ec/logos'),
      options: { up: 7 },
    },
    {
      from: [`${nodeModules}/@ecl/resources-eu-logo/dist/**/*.svg`],
      to: path.resolve(outputFolder, 'images/eu/logos'),
      options: { up: 7 },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-footer-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-footer-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-footer-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-footer-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
  ],
};
