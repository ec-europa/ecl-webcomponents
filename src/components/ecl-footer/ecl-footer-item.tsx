 import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'ecl-footer-item',
  shadow: false,
})

export class EclFooterItem{
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() link: string;
  @Prop() ariaLabel: string;

  getClass(): string {
    const styleClasses = [
      `ecl-site-footer__list-item`,
      `sc-ecl-footer-${this.theme}`,
      this.styleClass
    ];

    return styleClasses.join(' ');
  }

  render() { 
    return (
      <li
        class={this.getClass()}
      >
      <slot name="ecl-footer-item-desc"></slot>
        <ecl-link
          path={this.link}
          variant="standalone"
          styleClass={`ecl-site-footer__link sc-ecl-footer-${this.theme}`}
          theme={this.theme}
          ariaLabel={this.ariaLabel}
        >
          <slot></slot>  
        </ecl-link>
      </li>
    );
  }
}
