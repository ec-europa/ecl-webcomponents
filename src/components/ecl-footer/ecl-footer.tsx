import { Component, Prop, h, Element } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";

@Component({
  tag: 'ecl-footer',
  styleUrls: {
    ec: './build/styles/ecl-footer-ec.css',
    eu: './build/styles/ecl-footer-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclFooter {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';
  @Prop() logoAlt: string;
  @Prop() logoTitle: string;
  @Prop() logoLink: string;
  @Prop() logoAriaLabel: string;
  @Prop() logoLangCode: string;
  @Prop() siteName: string;
  @Prop() variant: string;
  @Prop() description: string;

  getClass(): string {
    const styleClasses = [
      `ecl-site-footer`,
      `ecl-site-footer--${this.variant}`,
      this.styleClass
    ];

    if (this.theme === 'ec' && this.variant === 'core') {
      styleClasses.push('ecl-site-footer--split-columns');
    }

    return styleClasses.join(' ');
  }

  componentDidLoad() {
    const lists = this.el.querySelectorAll('.ecl-site-footer__list');
    const titles = this.el.querySelectorAll('ecl-text[tag="h2"]');
    if (titles && titles[0]) {
      titles.forEach((title) => {
        const section = title.parentElement.parentElement;
        title.firstElementChild.classList.add(`sc-ecl-footer-${this.theme}`);
        section.insertBefore(title.firstElementChild, section.firstElementChild);
      });
    }
    if (lists) {
      lists.forEach((list) => {
        const items = list.querySelectorAll('.ecl-site-footer__list-item');
        if (items) {
          list.innerHTML = '';
          list.append(...items);
        }
      });
    }
  }

  render() {
    const logoPath = this.theme === 'ec' ?
      getAssetPath(`./build/images/${this.theme}/logos/negative/logo-${this.theme}--${this.logoLangCode}.svg`) :
      getAssetPath(`./build/images/${this.theme}/logos/standard-version/positive/logo-${this.theme}--${this.logoLangCode}.svg`);

    return (
      <footer
        class={this.getClass()}
      >
        <div class="ecl-container ecl-site-footer__container">
        { this.theme === 'eu' && this.variant === 'harmonised' ?
          <div class="ecl-site-footer__row">
            <div class="ecl-site-footer__column">
              <div class="ecl-site-footer__section  ecl-site-footer__section--site-info">
                <h2 class="ecl-site-footer__title">{this.siteName}</h2>
                <div class="ecl-site-footer__description">
                  {this.description}
                </div>
                <ul class={`ecl-site-footer__list`}>
                  <slot name="ecl-footer-list-info"></slot>
                </ul>
              </div>
            </div>
            <div class="ecl-site-footer__column">
              <div class="ecl-site-footer__section">
                <ul class="ecl-site-footer__list">
                  <slot name="ecl-footer-list-top-middle"></slot>
                </ul>
              </div>
              <div class="ecl-site-footer__section">
                <ul class="ecl-site-footer__list">
                  <slot name="ecl-footer-list-top-middle-bottom"></slot>
                </ul>
              </div>
            </div>
            <div class="ecl-site-footer__column">
              <div class="ecl-site-footer__section">
                <ul class="ecl-site-footer__list">
                  <slot name="ecl-footer-list-top-right"></slot>
                </ul>
              </div>
            </div>
          </div> : ''
        }
          <div class="ecl-site-footer__row">
            <div class="ecl-site-footer__column">
              <div class="ecl-site-footer__section">
              { this.variant === 'core' || this.variant === 'harmonised' ? 
                <ecl-link
                  path={this.logoLink}
                  theme={this.theme}
                  variant="standalone"
                  styleClass={`ecl-site-footer__logo-link sc-ecl-footer-${this.theme}`}
                  ariaLabel={this.logoAriaLabel}
                >
                  <ecl-picture
                    theme={this.theme}
                    imageAlt={this.logoAlt}
                    imgClass={`sc-ecl-footer-${this.theme} ecl-site-footer__logo-image`}
                    styleClass={`ecl-site-footer__picture sc-ecl-site-footer-${this.theme}`}
                    image={this.theme === 'ec' ? 
                      logoPath : 
                      getAssetPath(`./build/images/${this.theme}/logos/condensed-version/positive/logo-${this.theme}--${this.logoLangCode}.svg`)
                    }
                  >   
                    <source
                      srcSet={logoPath}
                      media="(min-width: 996px)"
                    ></source>
                    <slot name="sources"></slot>
                  </ecl-picture>
                </ecl-link> : ''
              }
              { this.theme === 'ec' ?
                <h2 class="ecl-site-footer__title">{this.siteName}</h2> : ''
              }
              { this.theme === 'ec' || this.variant === 'core' ? 
                <div class="ecl-site-footer__description">
                  <slot name="description"></slot>
                </div> :
                <div class="ecl-site-footer__description">
                  <ul class="ecl-site-footer__list">
                    <slot name="description"></slot>
                  </ul>
                </div>
              }
              { this.theme === 'ec' && this.variant !== 'core' ?
                <ul class="ecl-site-footer__list">
                  <slot name="ecl-footer-list-info"></slot>
                </ul> : '' 
              }
              </div>
            </div>
            <div class="ecl-site-footer__column">
              <div class={`ecl-site-footer__section ${this.theme === 'ec' && this.variant === 'core' ? 'ecl-site-footer__section--separator' : '' }`}>
                <ul class={`ecl-site-footer__list ${this.theme === 'ec' && this.variant === 'core' ? 'ecl-site-footer__list--columns' : '' }`}>
                  <slot name="ecl-footer-list-main"></slot>
                </ul>
              </div>
              <div class="ecl-site-footer__section">
                <ul class="ecl-site-footer__list">
                   <slot name="ecl-footer-list-bottom-left"></slot>
                </ul>
              </div> 
            { (this.variant === 'standardised' || this.variant === 'harmonised') ?
              <div class="ecl-site-footer__section">
                <ul class={`ecl-site-footer__list ${this.theme === 'ec' ? 'ecl-site-footer__list--inline' : '' }`}>
                  <slot name="ecl-footer-list-main-bottom"></slot>
                </ul>
              </div> : ''
            }
            { this.variant === 'core' ?
              <div class="ecl-site-footer__section">
                <ul class="ecl-site-footer__list">
                   <slot name="ecl-footer-list-bottom-right"></slot>
                </ul>
              </div> : ''
            }
            </div> 
            { this.theme === 'ec' && (this.variant === 'standardised' || this.variant === 'harmonised') ?
              <div class="ecl-site-footer__column">
                <div class="ecl-site-footer__section">
                  <ul class={`ecl-site-footer__list`}>
                    <slot name="ecl-footer-list-main-right"></slot>
                  </ul>
                </div>
                <div class="ecl-site-footer__section">
                  <ul class="ecl-site-footer__list">
                    <slot name="ecl-footer-list-main-right-bottom"></slot>
                  </ul>
                </div>
              </div> : ''
            }
            
          { this.theme === 'eu' ? 
            <div class="ecl-site-footer__column">
              <div class="ecl-site-footer__section">
                <ul class="ecl-site-footer__list">
                  <slot name="ecl-footer-list-right"></slot>
                </ul>
              </div>              
            </div> : ''
          }
          </div>
        { this.theme === 'ec' && (this.variant === 'standardised' || this.variant === 'harmonised') ?
            <div class="ecl-site-footer__row">
              <div class="ecl-site-footer__column">
                <div class="ecl-site-footer__section ecl-site-footer-section--condensed">
                  <div class="ecl-site-footer__content">More information on: </div>
                  <ul class="ecl-site-footer__list">
                    <slot name="ecl-footer-list-middle"></slot>
                  </ul>
                </div>
              </div>
            </div> : ''
          }
          { this.theme === 'ec' && (this.variant === 'standardised' || this.variant === 'harmonised') ? (
              <div class="ecl-site-footer__row">
                <div class="ecl-site-footer__column">
                  <div class="ecl-site-footer__section">
                    <ecl-link
                      theme={this.theme}
                      path={this.logoLink}
                      variant="standalone"
                      styleClass={`ecl-site-footer__logo-link sc-ecl-footer-${this.theme}`}
                      ariaLabel={this.logoAriaLabel}
                    >
                      <ecl-picture
                        theme={this.theme}
                        imageAlt={this.logoAlt}
                        imgClass={`ecl-site-footer__logo-image sc-ecl-site-footer-${this.theme}`}
                        image={logoPath}
                        styleClass={`ecl-site-footer__picture sc-ecl-site-footer-${this.theme}`}
                      >
                        <slot name="sources"></slot>
                      </ecl-picture>
                    </ecl-link>
                  </div>
                </div>
                <div class="ecl-site-footer__column">
                  <div class="ecl-site-footer__section ecl-site-footer__section--split-list">
                    <ul class="ecl-site-footer__list">
                      <slot name="ecl-footer-list-bottom-center"></slot>
                    </ul>
                  </div>
                </div>
                <div class="ecl-site-footer__column">
                  <div class="ecl-site-footer__section ecl-site-footer__section--split-list">
                    <ul class="ecl-site-footer__list">
                      <slot name="ecl-footer-list-bottom-right"></slot>
                    </ul>
                  </div>
                </div> 
              </div> ) : ( '' )
            }
        </div>
      </footer>
    )
  }
}

