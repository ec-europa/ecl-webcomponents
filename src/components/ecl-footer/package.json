{
  "name": "@ecl/web-components-footer",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.10.0",
  "description": "ECL Footer",
  "main": "ecl-hero-banner.tsx",
  "private": true,
  "scripts": {
    "build:copy": "cross-env ecl-builder copy",
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.9.1",
    "@ecl/resources-ec-logo": "3.9.1",
    "@ecl/resources-eu-logo": "3.9.1",
    "@ecl/vanilla-component-link": "3.9.1",
    "@ecl/vanilla-component-site-footer": "3.9.1",
    "@ecl/vanilla-layout-grid": "3.9.1",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
