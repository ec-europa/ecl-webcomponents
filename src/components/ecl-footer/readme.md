# ecl-footer



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type     | Default     |
| ------------ | ------------- | ----------- | -------- | ----------- |
| `ariaLabel`  | `aria-label`  |             | `string` | `undefined` |
| `link`       | `link`        |             | `string` | `undefined` |
| `styleClass` | `style-class` |             | `string` | `undefined` |
| `theme`      | `theme`       |             | `string` | `'ec'`      |


## Dependencies

### Depends on

- [ecl-link](../ecl-link)

### Graph
```mermaid
graph TD;
  ecl-footer-item --> ecl-link
  ecl-link --> ecl-icon
  ecl-link --> ecl-indicator
  style ecl-footer-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
