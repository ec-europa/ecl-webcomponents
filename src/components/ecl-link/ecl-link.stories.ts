import iconsAllEc from '@ecl/resources-ec-icons/dist/lists/all.json';

const getArgs = () => {
  return {
    variant: 'default',
    path: '/example',
    external: false,
    label: 'An ECL link',
    icon: '',
    iconPosition: 'after',
    iconTransform: '',
  };
};

const getArgTypes = () => {
  return {
    variant: {
      type: { name: 'select' },
      options: ['default', 'standalone', 'cta', 'primary', 'secondary', 'negative'],
      description: "Link variant"
    },
    label: {
      type: { name: 'string' },
      description: "Link label"
    },
    external: {
      type: { name: 'boolean' },
      description: "When the link is external"
    },
    path: {
      name: 'path/url',
      type: { name: 'string' },
      description: 'Path or Url to link to',
    },
    icon: {
      name: 'icon',
      type: { name: 'select' },
      options: iconsAllEc,
      description: "Add an icon",
    },
    iconPosition: {
      name: 'icon position',
      type: { name: 'select' },
      options: ['before', 'after'],
      description: 'Before or after the link label',
    },
    iconTransform: {
      name: 'icon transform',
      type: { name: 'select' },
      description: 'Link icon transform',
      options: [
        'rotate-90',
        'rotate-180',
        'rotate-270',
        'flip-horizontal',
        'flip-vertical',
      ],
    },
  };
};

export default {
  title: 'Components/Navigation/link',
};

const Template = (args) => 
  `<ecl-link
    variant="${args.variant}"
    theme="${args.theme}"
    path="${args.path}"
    external="${args.external}"
  >
  ${args.icon && args.iconPosition === 'before' ?
    `<ecl-icon 
      slot="icon-before"
      icon="${args.icon}"
      transform="${args.iconTransform}"
      theme="${args.theme}"
    >
    </ecl-icon>` : '' }
      ${args.label}
  ${args.icon && args.iconPosition === 'after' ?
    `<ecl-icon 
      slot="icon-after"
      icon="${args.icon}"
      theme="${args.theme}"
      transform="${args.iconTransform}"
    >
    </ecl-icon>` : '' }
  </ecl-link>`;

export const Link = Template.bind({});
Link.storyName = 'default';
Link.args = getArgs();
Link.argTypes = getArgTypes();
