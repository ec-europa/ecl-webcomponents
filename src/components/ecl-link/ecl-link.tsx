import { Component, h, Prop, Element} from '@stencil/core';

@Component({
  tag: 'ecl-link',
  styleUrls: {
    ec: './build/styles/ecl-link-ec.css',
    eu: './build/styles/ecl-link-eu.css',
  },
  shadow: false,
  scoped: false,
  assetsDirs: ['build'],
})

export class EclLink {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() path: string;
  @Prop() styleClass: string;
  @Prop() variant: string = 'default';
  @Prop() titleAttr: string;
  @Prop() external: boolean = false;

  getClass(): string {
    const styleClasses = [
      `ecl-link`,
      `ecl-link--${this.variant}`,
      this.styleClass
    ];

    if (this.el.querySelector('ecl-icon')) {
      styleClasses.push('ecl-link--icon');
      if (this.el.querySelector('ecl-icon').getAttribute('slot') === 'icon-after') {
        styleClasses.push('ecl-link--icon-after');
      } else {
        styleClasses.push('ecl-link--icon-before');
      }
    }

    return styleClasses.join(' ');
  }

  componentDidRender() {
    if (this.el.querySelector('ecl-icon')) {
      this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon');
      if (this.el.querySelector('ecl-icon').getAttribute('slot') === 'icon-after') {
        this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon--after');
      } else {
        this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon--before');
      }
    }
  }

  getExternal() {
    return (
      <ecl-icon 
        icon="external"
        style-class={`ecl-link__icon ecl-link__icon--after sc-ecl-link-${this.theme}`}
        size = "2xs"
      >
      </ecl-icon>
    )  
  }

  render() {
    return (
      <a
        class={this.getClass()}
        href={this.path}
        title={this.titleAttr}
      >
        <slot name="icon-before"></slot>
        <slot></slot>
        <slot name="icon-after">{ this.external ? this.getExternal() : '' }</slot>
      </a>
    );
  }
}
