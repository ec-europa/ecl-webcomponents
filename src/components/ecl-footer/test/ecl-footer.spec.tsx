import { newSpecPage } from '@stencil/core/testing';
import { EclFooter } from '../ecl-footer';
import { EclFooterItem } from '../ecl-footer-item';
import { EclLink } from '../../ecl-link/ecl-link';

describe('ecl-footer', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclFooter, EclFooterItem, EclLink],
      html: `<ecl-footer
        theme="ec"
        logo-alt="European Union"
        logo-title="European union"
        logo-lang-code="en"
        description="This site is managed by the Directorate-General for Communication"
        site-name="European Commission website"
      >
        <ecl-footer-item
          theme="ec"
          link="/example"
          slot="ecl-footer-list-main"
        >
          Strategy
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-main"
        >
          About the European Commission
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-main"
        >
          Business, Economy, Euro
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-main"
        >
          Live, work, travel in the EU
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-main"
        >
          Law
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-main"
        >
          Funding, Tenders
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-main"
        >
          Research and innovation
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-main"
        >
          Energy, Climate change, Environment
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-main"
        >
          Education
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-main"
        >
          Aid, Development cooperation, Fundamental rights
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-main"
        >
          Food, Farming, Fisheries
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-main"
        >
          EU regional and urban development
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-main"
        >
          Jobs at the European Commission
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-main"
        >
          Statistics
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-main"
        >
          News
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-main"
        >
          Events
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-main"
        >
          Publications
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-bottom-left"
        >
          About the European Commission's web presence
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-bottom-left"
        >
          Contact the European Commission
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-bottom-left"
        >
          Follow the European Commission on social media
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-bottom-left"
        >
          Resources for partners
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-bottom-right"
        >
          Languages on our websites
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-bottom-right"
        >
          Cookies
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-bottom-right"
        >
          Privacy policy
        </ecl-footer-item>
        <ecl-footer-item
          link="/example"
          theme="ec"
          slot="ecl-footer-list-bottom-right"
        >
          Legal notice
        </ecl-footer-item>
      </ecl-footer>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
