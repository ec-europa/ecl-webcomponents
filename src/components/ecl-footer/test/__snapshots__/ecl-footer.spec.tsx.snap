// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-footer renders 1`] = `
<ecl-footer description="This site is managed by the Directorate-General for Communication" logo-alt="European Union" logo-lang-code="en" logo-title="European union" site-name="European Commission website" theme="ec">
  <!---->
  <ecl-footer-item hidden="" link="/example.html" slot="ecl-footer-list-bottom-right" theme="ec">
    <!---->
    <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
      <ecl-link>
        <!---->
        <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
          Languages on our websites
          <slot-fb name="icon-after"></slot-fb>
        </a>
      </ecl-link>
    </li>
  </ecl-footer-item>
  <ecl-footer-item hidden="" link="/example.html" slot="ecl-footer-list-bottom-right" theme="ec">
    <!---->
    <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
      <ecl-link>
        <!---->
        <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
          Cookies
          <slot-fb name="icon-after"></slot-fb>
        </a>
      </ecl-link>
    </li>
  </ecl-footer-item>
  <ecl-footer-item hidden="" link="/example.html" slot="ecl-footer-list-bottom-right" theme="ec">
    <!---->
    <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
      <ecl-link>
        <!---->
        <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
          Privacy policy
          <slot-fb name="icon-after"></slot-fb>
        </a>
      </ecl-link>
    </li>
  </ecl-footer-item>
  <ecl-footer-item hidden="" link="/example.html" slot="ecl-footer-list-bottom-right" theme="ec">
    <!---->
    <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
      <ecl-link>
        <!---->
        <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
          Legal notice
          <slot-fb name="icon-after"></slot-fb>
        </a>
      </ecl-link>
    </li>
  </ecl-footer-item>
  <footer class="ecl-site-footer ecl-site-footer--undefined">
    <div class="ecl-container ecl-site-footer__container">
      <div class="ecl-site-footer__row">
        <div class="ecl-site-footer__column">
          <div class="ecl-site-footer__section">
            <h2 class="ecl-site-footer__title">
              European Commission website
            </h2>
            <div class="ecl-site-footer__description"></div>
            <ul class="ecl-site-footer__list"></ul>
          </div>
        </div>
        <div class="ecl-site-footer__column">
          <div class="ecl-site-footer__section">
            <ul class="ecl-site-footer__list">
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Strategy
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    About the European Commission
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Business, Economy, Euro
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Live, work, travel in the EU
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Law
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Funding, Tenders
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Research and innovation
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Energy, Climate change, Environment
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Education
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Aid, Development cooperation, Fundamental rights
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Food, Farming, Fisheries
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    EU regional and urban development
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Jobs at the European Commission
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Statistics
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Press corner
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Events
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Publications
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
            </ul>
          </div>
          <div class="ecl-site-footer__section">
            <ul class="ecl-site-footer__list">
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    About the European Commission's web presence
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Contact the European Commission
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Follow the European Commission on social media
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
              <li class="ecl-site-footer__list-item sc-ecl-footer-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--standalone ecl-site-footer__link sc-ecl-footer-ec" href="/example.html">
                    Resources for partners
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </footer>
</ecl-footer>
`;
